client/src/modules/shopping-list/model/actions.js

```diff
@@ -79,18 +80,42 @@ const fetchShoppingListsMetaDataRequest = () => ({
   type: ShoppingListActionTypes.FETCH_META_DATA_REQUEST
 });
 
+const archiveListSuccess = data => ({
+  type: ShoppingListActionTypes.ARCHIVE_SUCCESS,
+  payload: data
+});
+const archiveListFailure = errMessage => ({
+  type: ShoppingListActionTypes.ARCHIVE_FAILURE,
+  payload: errMessage
+});
+const archiveListRequest = () => ({
+  type: ShoppingListActionTypes.ARCHIVE_REQUEST
+});
+
+const restoreListSuccess = (data, listId) => ({
+  type: ShoppingListActionTypes.RESTORE_SUCCESS,
+  payload: { data, listId }
+});
+const restoreListFailure = errMessage => ({
+  type: ShoppingListActionTypes.RESTORE_FAILURE,
+  payload: errMessage
+});
+const restoreListRequest = () => ({
+  type: ShoppingListActionTypes.RESTORE_REQUEST
+});
+
 // Dispatchers
-export const fetchItemsFromGivenList = listId => dispatch => {
-  dispatch(fetchProductRequest());
-  getData(`${ENDPOINT_URL}/shopping-lists/${listId}/products`)
+export const fetchDataFromGivenList = listId => dispatch => {
+  dispatch(fetchDataRequest());
+  getData(`${ENDPOINT_URL}/shopping-lists/${listId}/data`)
```

Comment:
consider returning the result - maybe we will need that in the place where it is called in the future